### PR TITLE
FMT_VARIADIC_CONST - Support for const variadic methods

### PR DESCRIPTION
**Pull request by ludekvodicka**
Implementation of FMT_VARIDIC_CONST as discussed here https://github.com/fmtlib/fmt/issues/589

### CODE REVIEW

File: fmt/format.h [hunk 1]
```diff
@@ -3647,35 +3647,35 @@ void arg(WStringRef, const internal::NamedArg<Char>&) FMT_DELETED_OR_UNDEFINED;
 #else
 // Defines a wrapper for a function taking __VA_ARGS__ arguments
 // and n additional arguments of arbitrary types.
-# define FMT_WRAP(Char, ReturnType, func, call, n, ...) \
+# define FMT_WRAP(Const,Char, ReturnType, func, call, n, ...) \
```
Comment (vitaut) [inline fmt/format.h]: nit: missing space after `,`

File: fmt/format.h [hunk 2]
```diff
@@ -3647,35 +3647,35 @@ void arg(WStringRef, const internal::NamedArg<Char>&) FMT_DELETED_OR_UNDEFINED;
...
+      FMT_GEN(n, FMT_MAKE_ARG)) Const { \
     fmt::internal::ArgArray<n>::Type arr; \
     FMT_GEN(n, FMT_ASSIGN_##Char); \
     call(FMT_FOR_EACH(FMT_GET_ARG_NAME, __VA_ARGS__), fmt::ArgList( \
       fmt::internal::make_type(FMT_GEN(n, FMT_MAKE_REF2)), arr)); \
   }
 
-# define FMT_VARIADIC_(Char, ReturnType, func, call, ...) \
-  inline ReturnType func(FMT_FOR_EACH(FMT_ADD_ARG_NAME, __VA_ARGS__)) { \
+# define FMT_VARIADIC_(Const,Char, ReturnType, func, call, ...) \
```
Comment (vitaut) [inline fmt/format.h]: also here

File: fmt/format.h [hunk 3]
```diff
@@ -3706,10 +3706,16 @@ void arg(WStringRef, const internal::NamedArg<Char>&) FMT_DELETED_OR_UNDEFINED;
   \endrst
  */
 #define FMT_VARIADIC(ReturnType, func, ...) \
-  FMT_VARIADIC_(char, ReturnType, func, return func, __VA_ARGS__)
+  FMT_VARIADIC_(,char, ReturnType, func, return func, __VA_ARGS__)
```
Comment (vitaut) [inline fmt/format.h]: and here

**Comment by ludekvodicka** [comment]
Sure, done. Sorry for this, I overlooked it.

File: test/format-test.cc [hunk 4]
```diff
@@ -1630,6 +1630,24 @@ TEST(FormatTest, FormatMessageExample) {
...
+		w.write(format, args);
+		return w.str();
+	}
+	FMT_VARIADIC_CONST(std::string, format_message, int, const char *)
+};
+
+TEST(FormatTest, ConstFormatMessage) {
+	test_class c;
+	EXPECT_EQ("[42] something happened",
+		c.format_message(42, "{} happened", "something"));
```
Comment (vitaut) [inline test/format-test.cc]: nit: Please use 2-space indent instead of tabs.

Comment (ludekvodicka) [reply test/format-test.cc]: Done. It's interesting because it's copy-paste from your code. Visual studio had to reformat it.

**Comment by vitaut** [comment]
Merged, thanks!

**Comment by ludekvodicka** [comment]
You're welcome. And thank you for your patience, this was my first public PR ;-)

**Pull request merged**